src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -1066,3 +1083,125 @@ FIXTURE_TEST(check_segment_size_jitter, storage_test_fixture) {
         [size = sizes[0]](auto other) { return size == other; }),
       false);
 }
+
+FIXTURE_TEST(adjacent_segment_compaction, storage_test_fixture) {
```

Comment:
nit: would be good to add a recovery or part of the `op-fuzzer`  operation to ensure that crashing and removing things work.